pkg/controller/core.oam.dev/v1alpha2/core/components/componentdefinition/componentdefinition_controller.go

```diff
@@ -76,59 +73,23 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	if !r.matchControllerRequirement(&componentDefinition) {
-		klog.InfoS("skip componentDefinition: not match the controller requirement of componentDefinition", "componentDefinition", klog.KObj(&componentDefinition))
+	if !coredef.MatchControllerRequirement(&componentDefinition, r.controllerVersion, r.ignoreDefNoCtrlReq) {
+		klog.InfoS("skip definition: not match the controller requirement of definition", "componentDefinition", klog.KObj(&componentDefinition))
 		return ctrl.Result{}, nil
 	}
 
-	// refresh package discover when componentDefinition is registered
-	if componentDefinition.Spec.Workload.Type != types.AutoDetectWorkloadDefinition {
-		err := utils.RefreshPackageDiscover(ctx, r.Client, r.dm, r.pd, &componentDefinition)
```

Comment:
why RefreshPackageDiscover is unsafe?

Reply:
It affects performance and the code has been commented a long time.